config/main.py

```diff
@@ -1852,27 +1855,71 @@ def override_config_table(db, input_config_db, dry_run):
 
     config_db = db.cfgdb
 
+    # Read config from configDB
+    current_config = config_db.get_config()
+    # Serialize to the same format as json input
+    sonic_cfggen.FormatConverter.to_serialized(current_config)
+
+    updated_config = update_config(current_config, config_input)
+
+    yang_enabled = is_yang_config_verification_enabled(config_db)
+    if yang_enabled:
+        sy = sonic_yang_with_loaded_models(YANG_DIR)
+        # Validate running config
+        validate_config_by_yang(sy, current_config)
+        # Validate input config
+        validate_config_by_yang(sy, config_input)
+        # Validate updated whole config
+        validate_config_by_yang(sy, updated_config)
+
     if dry_run:
-        # Read config from configDB
-        current_config = config_db.get_config()
-        # Serialize to the same format as json input
-        sonic_cfggen.FormatConverter.to_serialized(current_config)
-        # Override current config with golden config
-        for table in config_input:
-            current_config[table] = config_input[table]
-        print(json.dumps(current_config, sort_keys=True,
+        print(json.dumps(updated_config, sort_keys=True,
                          indent=4, cls=minigraph_encoder))
     else:
-        # Deserialized golden config to DB recognized format
-        sonic_cfggen.FormatConverter.to_deserialized(config_input)
-        # Delete table from DB then mod_config to apply golden config
-        click.echo("Removing configDB overriden table first ...")
-        for table in config_input:
-            config_db.delete_table(table)
-        click.echo("Overriding input config to configDB ...")
-        data = sonic_cfggen.FormatConverter.output_to_db(config_input)
-        config_db.mod_config(data)
-        click.echo("Overriding completed. No service is restarted.")
+        override_config_db(config_db, config_input)
+
+
+def is_yang_config_verification_enabled(config_db):
```

Comment:
>is_yang_config_verification_enabled [](http://example.com/codeflow?start=4&length=35)

This may be reused by @isabelmsft. Find a better place?

Reply:
Draft a PR here. https://github.com/sonic-net/sonic-buildimage/pull/11715

Reply:
Can you calling function in that PR now?

Reply:
No. Current call that funtion will fail PR checker. Because the VM being tested now has not included that PR.